src/ProgressOnderwijsUtils/Data/CommandFactory.cs

```diff
@@ -39,7 +38,7 @@ public struct ReusableCommand : IDisposable
         public void Dispose()
         {
             QueryTimer?.Dispose();
-            if (Command != null) {
+            if (Command.PretendNullable() != null) {

```

Comment:
Normaal gesproken zou jij vinden dat je Command nullable moet maken toch?

Reply:
Het objectje is echt non-nullable; echter vanwege value-type semantics is het denkbaar dat iemand iets stouts doet - en in een dispose wil ik niet crashen dan; dat is heel vervelend.

Als ik em nullable maak, dan "klopt" de annotatie voor de post-dispose use case; behalve natuurljik dat je dan alsnog het niet mag gebruiken, dus echt kloppen doet dat niet; dat soort typing heeft C# niet (post-dispose is dit meer `void` dan iets anders).

Maar indien het nullable zou worden klopt ie minder pre-dispose; dan moet je gewoon overal AssertNotNull doen.

Ik ben niet fan van dit soort gerommel, maar ik ben dan ook niet fan van custom IDisposable implementeren ;-); maar ik denk dat het soms nodig is, en hier een meerwaarde heeft. Ik weet iig geen oplossing hier die goed werkt.

Ik zou zelfs `null!` hier prima vinden.

Reply:
Ik zou nog evt. Command kunnen dupliceren en een non-nullable accessor met assert maken, en een "interne" die nullable is.  Dat zal heel klein beetje extra overhead hebben, maar vast niet boeiend?

Reply:
zo?